library/src/schemas/bigint/bigint.ts

```diff
@@ -36,28 +40,13 @@ export function bigint(
   arg2?: Pipe<bigint>
 ): BigintSchema {
   // Get error and pipe argument
-  const [error, pipe] = getDefaultArgs(arg1, arg2);
+  const [error, pipe = []] = getDefaultArgs(arg1, arg2);
 
   // Create and return bigint schema
   return {
-    /**
-     * The schema type.
-     */
-    schema: 'bigint',
-
-    /**
-     * Whether it's async.
-     */
+    kind: 'bigint',
```

Comment:
Would you prefer `kind` to `type`? If so, for what reason?

Reply:
If think that `type` is more generic and also reduces the bundle size a bit. However, the bundle size is negligible here. I am mainly interested in good naming.

Reply:
I think `kind` would be better so that it doesn't have a potential conflict if someone is trying to build out a JSON schema library on top of this.

It's also what I've seen commonly used to differentiate similar types in discriminated unions, which is basically what we have here to refer to all schemas/validations/transforms that extend a common base class.

Reply:
If anything, this would be better read as `type: "schema", kind: "bigint"` for example. Where `type` refers to the base type we're extending from, and `kind` refers to the sub-type.

Reply:
I understand the JSON argument, but would not consider it for Valibot as it seems too irrelevant at the moment. I think `kind` is fine and am happy with either name. I just thought `type` was more commonly used for differentiation. We can continue with `kind` and update if needed before v1 if we have better ideas or arguments.

Reply:
> If anything, this would be better read as `type: "schema", kind: "bigint"` for example. Where `type` refers to the base type we're extending from, and `kind` refers to the sub-type.

That's a good argument!

Reply:
I have been reading about the difference between `kind` and `type`. Several sources have indicated that `kind` refers to a group with certain attributes, while `type` refers to a specific item. So I prefer to continue with `type`.